pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -972,4 +1017,41 @@ public void onUpdate(String path, Map<String, String> data, Stat stat) {
         return dynamicConfigurationMap;
     }
 
-}
+    /**
+     * Create pending topic and on completion it picks the next one until processes all topics in
+     * {@link #pendingTopicLoadingQueue}.<br/>
+     * It also tries to acquire {@link #topicLoadRequestSemaphore} so throttle down newly incoming topics and release
+     * permit if it was successful to acquire it.
+     */
+    private void createPendingLoadTopic() {
+        Pair<String, CompletableFuture<Topic>> pendingTopic = pendingTopicLoadingQueue.poll();
+        if (pendingTopic != null) {
+            final String topic = pendingTopic.getLeft();
+            try {
+                checkTopicNsOwnership(topic);
+                CompletableFuture<Topic> future = new CompletableFuture<>();
+                final Semaphore topicLoadSemaphore = topicLoadRequestSemaphore.get();
+                final boolean acquiredPermit = topicLoadSemaphore.tryAcquire();
```

Comment:
What is the reason for acquiring the semaphore here, if we're not waiting for it anyway?